modules/energid/collect.go

```diff
@@ -0,0 +1,161 @@
+package energid
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/netdata/go.d.plugin/pkg/stm"
+	"github.com/netdata/go.d.plugin/pkg/web"
+)
+
+const (
+	urlPathStat = "/"
+)
+
+func (e *Energid) collect() (map[string]int64, error) {
+	ms := e.scrapeEnergid()
+
+	if ms == nil {
+		return nil, fmt.Errorf("Invalid request")
+	}
+
+	collected := make(map[string]int64)
```

Comment:
i think you just need to set `stm` tags you want and then stm.ToMap energidStats and not specific parts of it